app/uk/gov/hmrc/softdrinksindustrylevystub/models/CreateSubscription.scala

```diff
@@ -0,0 +1,145 @@
+  ProducedHigher: Option[Int],
+  ImportedLower: Option[Int],
+  ImportedHigher: Option[Int],
+  PackagedLower: Option[Int],
+  PackagedHigher: Option[Int]
+)
+
+case class BankDetails (
+  directDebit: Boolean,
+  accountName: Option[String],
+  accountNumber: Option[String],
+  sortCode: Option[String],
+  buildingSocietyRollNumber: Option[String]
+)
+
+object ProducerClassification extends Enumeration {
+  type ProducerClassification = Value
+  val Unknown, Large, Small = Value
+  implicit val producerClassificationFormat = EnumUtils.enumFormat(ProducerClassification)
+}
+
+
+object OrganisationType extends Enumeration {
+  type OrganisationType = Value // TODO - figure out if these type statements are needed
+  val Unknown, SoleProprietor, LimitedCompany, LLP, UnincorporatedBody, Partnership, Trust = Value
+  implicit val organisationTypeFormat = EnumUtils.enumFormat(OrganisationType)
+}
+
+object ActionType extends Enumeration {
+  val Unknown, Add, Amend, Remove = Value
+  implicit val actionTypeFormat = EnumUtils.enumFormat(ActionType)
+}
+
+object EntityType extends Enumeration {
+  val Unknown, GroupMember, GroupRepresentativeMember, ControllingBody, Partner = Value
+  implicit val entityTypeFormat = EnumUtils.enumFormat(EntityType)
+}
+
+object ActivityType extends Enumeration {
+  val Unknown, Producer, Importer, ContractPacker = Value
+  implicit val activityTypeFormat = EnumUtils.enumFormat(ActivityType)
+}
+case class LevyDetails(
+  activities: ActivityType.Value,
+  lessThanMillion: Boolean,
+  producerClassification: Option[ProducerClassification.Value],
+  smallProducerExemption: Boolean,
+  usesCopacker: Boolean,
+  voluntarilyRegistered: Boolean
+)
+
+object SiteAction extends Enumeration {
+  val Unknown, NewSite, AmendSite, CloseSite, TransferSite = Value
+  implicit val siteActionFormat = EnumUtils.enumFormat(SiteAction)
+}
+
+object SiteType extends Enumeration {
+  val Unknown, Warehouse, ProductionSite = Value
+  implicit val siteTypeFormat = EnumUtils.enumFormat(SiteType)
+}
+
+case class Site (
+  action: Option[SiteAction.Value],
+  siteReference: Option[String],
+  dateOfClosure: Option[Date],
+  siteClosureReason: Option[String],
+  tradingName: Option[String],
+  newSiteReference: Option[String],
+  address: Address,
+  typeOfSite: Option[SiteType.Value]
+)
+
+case class CreateSubscriptionRequest (
+  organisationType: OrganisationType.Value,
+  action: Option[ActionType.Value],
+  typeOfEntity: Option[EntityType.Value],
+  dateOfApplication: Date,
+  taxStartDate: Date,
+  joiningDate: Option[Date],
+  leavingDate: Option[Date],
+  customerIdentificationNumber: String,
+  tradingName: String,
+  businessContactDetails: Address,
+  correspondenceAddressDiffers: Boolean,
+  correspondenceAddress: Option[Address],
+  primaryPerson: ContactDetails,
+  softDrinksIndustryLevyDetails: LevyDetails,
+  sdilActivity: LitresProduced,
+  estimatedAmountOfTaxInTheNext12Months: Option[Int], // TODO change this to BigDecimal
```

Comment:
Do we need to change this to big decimal now or later on?

Reply:
later.